packages/slo-stat-bot/test/slo-stat-label.ts

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import {resolve} from 'path';
+import {Probot} from 'probot';
+import nock from 'nock';
+import * as fs from 'fs';
+import {expect} from 'chai';
+import {describe, it, beforeEach} from 'mocha';
+import Webhooks from '@octokit/webhooks';
+import snapshot from 'snap-shot-it';
+
+import handler from '../src/slo-stat-label';
+import spies from 'chai-spies';
+import { doesNotReject } from 'assert';
+
+const chai = require('chai');
+chai.use(spies);
+const sinon = require('sinon');
+
+
+nock.disableNetConnect();
+
+const fixturesPath = resolve(__dirname, '../../test/fixtures');
+
+describe('slo-status-label', () => {
+  let probot: Probot;
+
+  beforeEach(() => {
+    probot = new Probot({
+      // use a bare instance of octokit, the default version
+      // enables retries which makes testing difficult.
+      // eslint-disable-next-line node/no-extraneous-require
+      Octokit: require('@octokit/rest'),
+    });
+
+    probot.app = {
+      getSignedJsonWebToken() {
+        return 'abc123';
+      },
+      getInstallationAccessToken(): Promise<string> {
+        return Promise.resolve('abc123');
+      },
+    };
+    probot.load(handler);
+  });
+
+  describe('opened or reopened pull request', () => {
+    let payload: Webhooks.WebhookPayloadPullRequest;
+    let stub: any;
```

Comment:
As opposed to providing the `any` type assign it to `sinon.SinonStub`

Reply:
I created a sandbox instead so I don't have to keep track of any fakes or stubs in case I make more for future test cases. I was unable to set the return type to be specifically sinon.SinonStub because it was erroring out.